src/configuration_api.rs

```diff
@@ -0,0 +1,62 @@
+// Built-in uses
+use std::sync::{Arc, RwLock};
+
+// External uses
+use jsonrpc_core::Result;
+use jsonrpc_derive::rpc;
+
+// Workspace uses
+
+// Local uses
+use crate::{node::InMemoryNodeInner, ShowCalls};
+
+pub struct ConfigurationApiNamespace {
+    node: Arc<RwLock<InMemoryNodeInner>>,
+}
+
+impl ConfigurationApiNamespace {
+    pub fn new(node: Arc<RwLock<InMemoryNodeInner>>) -> Self {
+        Self { node }
+    }
+}
+
+#[rpc]
+pub trait ConfigurationApiNamespaceT {
+    #[rpc(name = "config_getShowCalls", returns = "String")]
+    fn config_get_show_calls(&self) -> Result<String>;
+    
+    #[rpc(name = "config_setShowCalls", returns = "String")]
+    fn config_set_show_calls(&self, value: String) -> Result<String>;
+
+    #[rpc(name = "config_setResolveHashes", returns = "bool")]
+    fn config_get_resolve_hashes(&self, value: bool) -> Result<bool>;
+}
+
+impl ConfigurationApiNamespaceT for ConfigurationApiNamespace {
+    fn config_get_show_calls(&self) -> Result<String> {
+        let reader = self.node.read().unwrap();
+        Ok(reader.show_calls.to_string())
+    }
+
+    fn config_set_show_calls(&self, value: String) -> Result<String> {
+        let show_calls = ShowCalls::from_str(&value);
+        
+        match show_calls {
```

Comment:
you can refactor this a little:
```
let mut inner = self.nolde.write().unwrap();
if let Some(show_calls) = show_calls {
  inner.show_calls = show_calls;
} 
Ok(reader.show_calls.to_string());
```

Reply:
Oh nice!! I tried to do something similar but having the reader in the same path as the write caused errors with Rust.

Reply:
Yup, this runs into the same deadlock error as I hit before:
`hread 'tokio-runtime-worker' panicked at 'rwlock read lock would result in deadlock'`